### PR TITLE
minor code cleanup in Curl_http_readwrite_headers()

### PR DESCRIPTION
**Pull request by mkauf**
The ``if(!*stop_reading)`` is strange because the code inside the ``if`` does not read data... and the condition of the ``if`` is always true.

### CODE REVIEW

File: lib/http.c [hunk 1]
```diff
@@ -3054,7 +3054,7 @@ CURLcode Curl_http_readwrite_headers(struct SessionHandle *data,
         switch(k->httpcode) {
         case 100:
           /* if we did wait for this do enable write now! */
-          if(k->exp100) {
+          if(k->exp100 > EXP100_SEND_DATA) {
```
Comment (bagder) [inline lib/http.c]: This looks like an unrelated change. Right?

Comment (mkauf) [inline lib/http.c]: It's just a style change... now it's the same as in transfer.c line 449

Of course you may skip this change if you want.

**Comment by bagder** [comment]
Good catch. Your finding made me look closer at this function.

That code uses a do-while loop that uses the condition `while(!*stop_reading && *k->str);` [here](https://github.com/curl/curl/blob/master/lib/http.c#L3764). But studying the code I can't find anyway where `*stop_reading` would be set to `TRUE` without a following unconditional break (out from the do-while loop), thus making the condition in the while totally pointlessly checking the `*stop_reading` value. If so, we could also remove that always constant condition from the while. You agree?

File: lib/http.c (hunk 1 above)
Comment (bagder) [inline lib/http.c]: Oh, right. Consistency is good too...

**Comment by mkauf** [comment]
Yes, I agree :-) Thank you for the thorough review!

**Comment by mkauf** [comment]
I have updated the pull request (removed ``!*stop_reading`` from the while condition)

**Comment by bagder** [comment]
thanks!
